### PR TITLE
Listing

### PR DESCRIPTION
**Pull request by elishkaflint**

### CODE REVIEW

File: lib/listing.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  def self.all
+    Listing.switch_database
+    result = @connection.exec("SELECT * FROM listings")
+    result.map { |listing| Listing.new(listing['id'], listing['name']) }
+  end
+
+  private
+
+  def self.switch_database
+    if ENV['RACK_ENV'] == 'test'
```
Comment (danieljonesdmj) [inline lib/listing.rb]: should the ENV be environment instead?

File: lib/listing.rb [hunk 2]
```diff
@@ -22,7 +22,7 @@ def self.all
   private
 
   def self.switch_database
-    if ENV['RACK_ENV'] == 'test'
+    if ENV['ENVIRONMENT'] == 'test'
```
Comment (elishkaflint) [inline lib/listing.rb]: Updated environment variable

File: Rakefile [hunk 3]
```diff
@@ -17,7 +17,7 @@ task :setup_databases do
     # when we know what database tables we want:
     connection = PG.connect(dbname: database)
     connection.exec("CREATE TABLE users (id SERIAL PRIMARY KEY, username VARCHAR(60), password VARCHAR(60));")
-    # connection.exec("CREATE TABLE table_name (id SERIAL PRIMARY KEY, user_id INTEGER REFERENCES users(id) ON DELETE CASCADE, content VARCHAR(240), time VARCHAR(60));")
```
Comment (laythq) [inline Rakefile]: This will probably conflict with our merge, but can be resolved easily.

File: spec/listing_spec.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  end
+
+  describe ".name" do
+    it "shows the name of a listing" do
+      listing = Listing.create("name")
+      expect(listing.name).to eq ("name")
+    end
+  end
+
+end
```
Comment (laythq) [inline spec/listing_spec.rb]: :shipit:

**Pull request merged**